### PR TITLE
[ADD] estate: Apply constraints to selling price and other fields T#70778

### PR DESCRIPTION
**Pull request by Gilmaciasr**
Add SQL constraints to Tag and Type to avoid duplicate names, for selling_price, expected_price and offers constraints were applied to avoid assigning a negative value, the same was established for living area and garden area to avoid a total_area result of 0 or negative values.
Create a constraint to accept an offer only if its price is at least 90% of the expected price, for that the accept_offer_button_action
had to be changed in order to trigger the selling price update and therefore apply the constraint.
Link to the lesson:
https://www.odoo.com/documentation/15.0/developer/tutorials/getting_started/11_constraints.html
Video of current changes:
https://youtu.be/NuDJWH2toOA

### CODE REVIEW

**Comment by Gilmaciasr** [comment]
@JesusValdez96  Ready PR#5! Can you please check it out?

File: estate/models/estate_property_offer.py [hunk 1]
```diff
@@ -40,7 +44,13 @@ def accept_offer_button_action(self):
...
                 record.status = "accepted"
+                self.property_id.selling_price = record.price
+                self.property_id.buyer_id = record.partner_id
+        return True
 
     def refuse_offer_button_action(self):
         for record in self:
             record.status = "refused"
+            self.property_id.selling_price = 0
+            self.property_id.buyer_id = None
```
Comment (JesusValdez96) [inline estate/models/estate_property_offer.py:55]:
```suggestion
            record.property_id.selling_price = 0
            record.property_id.buyer_id = False
```

File: estate/models/estate_property_offer.py [hunk 2]
```diff
@@ -40,7 +44,13 @@ def accept_offer_button_action(self):
         for record in self:
             if "accepted" not in record.property_id.offer_ids.mapped("status"):
                 record.status = "accepted"
+                self.property_id.selling_price = record.price
+                self.property_id.buyer_id = record.partner_id
```
Comment (JesusValdez96) [inline estate/models/estate_property_offer.py:48]:
```suggestion
                record.property_id.selling_price = record.price
                record.property_id.buyer_id = record.partner_id
```

**Review by JesusValdez96 — CHANGES_REQUESTED** [review]

**Comment by Gilmaciasr** [comment]
@JesusValdez96  suggested changes done!

**Pull request merged**